### DISCUSSION TITLE
Rename `onMutate` to `onBeforeMutate`

### DISCUSSION THREAD

**Discussion by jd-solanki** [posva/pinia-colada#104] [Ideas]
Hi 👋🏻 

I was awake late night to just wathc yours and server component talk. Nice talk, I will try pinia-colada in my current project.

During talk I realized it'll be better if we rename `onMutate` to `onBeforeMutate` to clearly indicate that this will run before the mutation and not after. Seeing `onMutate` raises question "Will it run before or after".

**Comment by posva** [comment]
Thanks, I'm glad you liked my talk!

`onMutate()` is still correct as it triggers when mutate is called. Triggering after is not as useful, so I prefer to keep this wording. On top of that, other libraries like tanstack query also use `onMutate()`, so this make skill transfer easier 🙂

**Reply by posva** [reply]
On second thought, I think it makes it more intuitive to be named `onBeforeMutate()`. I will make `onMutate` an alias for migration purposes

**Reply by jd-solanki** [reply]
Thanks for the change!

I appreciate that you came back and commented on it 👏🏻

**Reply by posva** [reply]
I researched a bit more and it turns out the current naming with just `on<event>` aligns better with the existing usage in the JS world as it means "run this as mutate is about to happen, or is happening". Running after wouldn't make sense since all the other hooks are about running after and in all scenarios, you can still change the outcome, for example, throwing during `onMutate()` will cancel the mutation, similar to calling `event.preventDefault()`
